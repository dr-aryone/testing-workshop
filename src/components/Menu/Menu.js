import React from "react";
import { Link } from "react-router-dom";

const Menu = () => {
  return (
    <header data-testid="menuContainer">
      <h1>Menu</h1>
      <Link to="/toggle">Toggle</Link>
      <Link to="/change">Change Name</Link>
      <Link to="/counter">Counter</Link>
    </header>
  );
};

export default Menu;
