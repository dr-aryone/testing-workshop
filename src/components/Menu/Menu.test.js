import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import Menu from "./Menu";

const renderWithRouter = component =>
  render(<BrowserRouter>{component}</BrowserRouter>);

describe("Menu Page", () => {
  it("Renders without crashing", () => {
    ReactDOM.render(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>,
      document.createElement("div")
    );

    ReactDOM.unmountComponentAtNode(document.createElement("div"));
  });

  it("Loads component", () => {
    const { getByTestId } = renderWithRouter(<Menu />);

    const menu = getByTestId("menuContainer");

    expect(menu).toBeInTheDocument();
  });
});
