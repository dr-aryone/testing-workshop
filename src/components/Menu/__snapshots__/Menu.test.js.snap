// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Page Match Snapshot 1`] = `undefined`;
